engine/access/rpc/backend/scripts/executor/failover.go

```diff
@@ -2,37 +2,46 @@ package executor
 
 import (
 	"context"
-	"time"
 
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
+	"github.com/onflow/flow-go/access"
+	accessmodel "github.com/onflow/flow-go/model/access"
+	"github.com/onflow/flow-go/module/executiondatasync/optimistic_sync"
 )
 
+// FailoverScriptExecutor is a script executor that executes scripts and gets accounts using local storage first,
+// then falls back to execution nodes if data is not available for the height or if request
+// failed due to a non-user error.
 type FailoverScriptExecutor struct {
 	localExecutor         ScriptExecutor
 	executionNodeExecutor ScriptExecutor
 }
 
 var _ ScriptExecutor = (*FailoverScriptExecutor)(nil)
 
+// NewFailoverScriptExecutor creates a new [FailoverScriptExecutor].
 func NewFailoverScriptExecutor(localExecutor ScriptExecutor, execNodeExecutor ScriptExecutor) *FailoverScriptExecutor {
 	return &FailoverScriptExecutor{
 		localExecutor:         localExecutor,
 		executionNodeExecutor: execNodeExecutor,
 	}
 }
 
-func (f *FailoverScriptExecutor) Execute(ctx context.Context, request *Request) ([]byte, time.Duration, error) {
-	localResult, localDuration, localErr := f.localExecutor.Execute(ctx, request)
-
-	isInvalidArgument := status.Code(localErr) == codes.InvalidArgument
-	isCanceled := status.Code(localErr) == codes.Canceled
-	if localErr == nil || isInvalidArgument || isCanceled {
-		return localResult, localDuration, localErr
+// Execute executes the provided script at the requested block.
+//
+// Expected error returns during normal operation:
+//   - [access.InvalidRequestError] - if the script execution failed due to invalid arguments or runtime errors.
+//   - [access.RequestCanceledError] - if the script execution was canceled.
+//   - [access.DataNotFoundError] - if the data was not found.
+//   - [access.ServiceUnavailable] - if no nodes are available or a connection to an execution node could not be established.
+//   - [access.InternalError] - if an internal failure occurs.
```

Comment:
this has fewer errors listed than in the interface. since this is potentially calling both the local and en providers, shouldn't it have the full set?